### PR TITLE
Add ADCC metric

### PR DESCRIPTION
**Pull request by GalyaZalesskaya**
- Added ADCC metric

image
image

### CODE REVIEW

File: openvino_xai/metrics/adcc.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        :return: A dictionary containing the average ADCC score.
+        :rtype: Dict[str, float]
+        """
+        results = []
+        for input_image, explanation in zip(input_images, explanations):
+            for class_idx, saliency_map in explanation.saliency_map.items():
+                metric_dict = self(saliency_map, int(class_idx), input_image)
+                results.append(metric_dict["adcc"])
+        adcc = np.mean(np.array(results), axis=0)
+        return {"adcc": adcc}
```
Comment (goodsong81) [inline openvino_xai/metrics/adcc.py:132]: Is it unusual to return average of other sub metrics like coherency?

File: openvino_xai/metrics/adcc.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        Returns:
+        :return: A dictionary containing the average ADCC score.
+        :rtype: Dict[str, float]
+        """
+        results = []
+        for input_image, explanation in zip(input_images, explanations):
+            for class_idx, saliency_map in explanation.saliency_map.items():
+                metric_dict = self(saliency_map, int(class_idx), input_image)
+                results.append(metric_dict["adcc"])
+        adcc = np.mean(np.array(results), axis=0)
```
Comment (goodsong81) [inline openvino_xai/metrics/adcc.py:131]: I'm not sure which is right, but we could alternatively compute ADCC based on the averages of sub metrics.
Simply put, average of ADCC vs ADCC of average of sub metrics.
Mark implemented like the latter approach for the dataset evaluation.
There is no reference for multi-image evaluation in the original repo, unfortunately.

File: openvino_xai/metrics/adcc.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        """
+
+        masked_image = image * saliency_map[:, :, None]
+        saliency_map_mapped_image = self.explainer(masked_image, targets=[class_idx], colormap=False, scaling=False)
+        saliency_map_mapped_image = saliency_map_mapped_image.saliency_map[class_idx]
+
+        A, B = saliency_map, saliency_map_mapped_image
+        # Pearson correlation coefficient
+        Asq, Bsq = A.flatten(), B.flatten()
+        y, _ = STS.pearsonr(Asq, Bsq)
```
Comment (goodsong81) [inline openvino_xai/metrics/adcc.py:66]: These are from some convention?
What does `sq` mean?
STS? Can we just import `pearsonr`?
Please ignore if you think the code readers would likely be OK.

**Review by goodsong81 — APPROVED** [review]
Thank you for the update. I left some question-like comments below :)

File: openvino_xai/metrics/adcc.py (hunk 1 above)
Comment (GalyaZalesskaya) [reply openvino_xai/metrics/adcc.py:132]: Actually, we can do it as well. In ReciproCAM paper all averaged metrics are published.

File: openvino_xai/metrics/adcc.py (hunk 2 above)
Comment (GalyaZalesskaya) [reply openvino_xai/metrics/adcc.py:131]: Thank you for this point. The metric should be the same in both cases and the performance difference should not be visible. 

I personally think that it's better to average the adcc so `__call__` method will return `adcc` for one saliency map as well. I don't know why (and if) we need that, but I tried to keep the similarity between` __call__` and `evaluate` methods

File: openvino_xai/metrics/adcc.py (hunk 3 above)
Comment (GalyaZalesskaya) [reply openvino_xai/metrics/adcc.py:66]: I follow the original implementation from here https://github.com/aimagelab/ADCC/blob/main/metrics/coherency.py

Good point with importing just `pearsonr`, I'll update that and naming in the next PR.

File: openvino_xai/metrics/adcc.py (hunk 1 above)
Comment (GalyaZalesskaya) [reply openvino_xai/metrics/adcc.py:132]: I'll update it during refactoring

**Pull request merged**